packages/expo-errors/src/__tests__/Errors-test.ts

```diff
@@ -8,3 +8,71 @@ describe('UnavailabilityError', () => {
     expect(err.message).toContain('someProperty');
   });
 });
+
+describe('warnDeprecated', () => {
+  const originalWarning = console.warn;
+  function isWarnedAsync() {
+    return new Promise((resolve, reject) => {
+      console.warn = (...args: any[]) => {
+        resolve(args);
+        // return originalWarning(...args);
+      };
+    });
+  }
+  beforeEach(() => {
+    console.warn = originalWarning;
+  });
+  afterAll(() => {
+    console.warn = originalWarning;
+  });
+
+  it('warns of a future replacement', async () => {
```

Comment:
These tests don't seem to add anything over the ones without the setTimeout other than that they test that console.warn is called. The setTimeout part seems unnecessary though, and it seems a bit sketch that `isWarnedAsync` causes console.warn to be overridden, rather than setting up a mock function for console.warn at the beginning of the test. Something like this is more straightforward:

```js
      console.warn = jest.fn();
      Errors.warnDeprecated('expo-errors', 'foo', {
        replacement: 'bar',
        currentVersion: '1.0.0',
        versionToRemove: '2.0.0',
      });
      expect(console.warn).toHaveBeenCalledTimes(1);
      expect(console.warn.mock.calls[0]).toMatchSnapshot();  // or something else that makes sure the warning is as expected
```